doc/source/Container.md

```diff
@@ -0,0 +1,132 @@
+
+* Create the Watson Conversation service:
+
+```
+$ bx service create conversation free $CONVERSATION_SERVICE
+```
+
+
+* Verify that the service instance is created:
+
+```
+$ bx service list
+```
+
+* Obtain the ID of your cluster:
+
+```
+$ bx cs clusters
+```
+
+* Bind the service instance to your cluster:
+
+```
+$ bx cs cluster-service-bind <cluster-ID> $CONVERSATION_SERVICE
+```
+
+## Load the Watson Conversation
+
+* Follow the instructions to [Load the Watson Conversation](https://github.com/IBM/watson-conversation-slots-intro#4-configure-watson-conversation)
+
+## Create a Kubernetes Configuration Map with the Workspace ID
+
+* Create a Kubernetes Configuration Map:
+
+```
+$ kubectl create configmap $CONFIG_MAP \
+    --from-literal=workspace_id=<WorkSpace_ID>
+```
+
+* Verify that the configuration is set:
+
+```
+$ kubectl get configmaps $CONFIG_MAP -o yaml
+```
+
+## Deploy the Pod
+
+* Deploy
+
+```
+$ kubectl create -f $POD_NAME
+```
+
+* Identify the **Public IP** address of your worker:
+
+```
+$ bx cs workers $CLUSTER_NAME
+``` 
+
+* Identify the external port your pod is listening on. 
+
+>> Note: The pizza-bot container listens on port 3000. Kubernetes maps this a publicly addressable port:
+
+```
+$ kubectl get services $KUBE_SERVICE
+```
+
+* Access the application using `http://<IP Address>:<Port>`
+
+## Look Under the Hood
+
+Now that you have created and bound a service instance to your cluster, let's take a deeper look at what is happening behind the scenes.
+
+
+* Binding the conversation service to your cluster creates a Kubernetes secret named `binding-${CONVERSATION_SERVICE}` .
+
+* The secret contains a key named binding with its data being a JSON string of the form 
+
+```
+{"url":"https://gateway.watsonplatform.net/conversation/api",
+ "username":"service-instance-user-uuid",
+ "password":"service-instance-password"}.
+```
+
+The secret is mapped into the container as an environment variable through the Kubernetes pod configuration.
+
+```
+          - name: $CONTAINER_ENV_VARIABLE
+            valueFrom:
+              secretKeyRef:
+                name: binding-${CONVERSATION_SERVICE}
+                key: binding
+```
+
+* For an application that expects the service credentials to be set in the environment variables `CONVERSATION_USERNAME` and `CONVERSATION_PASSWORD`, we set these in the container environment. 
+
+* Example usage is to have a script that is run when the container is started. We can then parse them from the  environment variable `$CONTAINER_ENV_VARIABLE` using jq :
+
+```
+export CONVERSATION_USERNAME=$(echo "${CONVERSATION_ENV_VARIABLE}" |
+                                jq -r '.username')
+
+```
+
+You can see the defined secrets in the Kubernetes dashboard by running ``kubctrl proxy`` and accessing `http://127.0.0.1:8001/ui`.
```

Comment:
maybe use a link for that url so you can click and go?

Reply:
good idea